fix: [#2] addIfAbsent must resolve module before comparison

moarcaffeine opened the pull request:
Current check is comparing names to resolved modules. Fix for issue #2.

moarcaffeine committed:
fix: resolve module before comparison

Current check is comparing names to resolved modules.

davidje13 commented:
Interesting that I haven't seen the bug you describe myself (maybe it's OS-specific or only manifests for projects in paths with capital letters, etc.). This change looks fine, but 2 comments:

1. Is `includes` still needed, or is straightforward equality checking sufficient?
2. I assume `require.resolve` is somewhat expensive, so can you pull it out into a variable to avoid computing it for each check? (if it does its own internal caching then nevermind)

I can merge and push a new release tomorrow.

davidje13 committed:
Use simpler comparison for resolved paths

Use full string equality rather than partial matches, and resolve each path only once.
